scripts/sct_analyze_lesion.py

```diff
@@ -406,35 +401,39 @@ def _normalize(self, vect):
-            centerline = Centerline(x_centerline_fit, y_centerline_fit, z_centerline, x_centerline_deriv, y_centerline_deriv, z_centerline_deriv)
-
-            # average centerline coordinates over slices of the image
-            x_centerline_deriv_rescorr, y_centerline_deriv_rescorr, z_centerline_deriv_rescorr = centerline.average_coordinates_over_slices(im_seg)[3:]
-
-            # compute Z axis of the image, in physical coordinate
-            axis_Z = im_seg.get_directions()[2]
-
-            # for iz in range(min_z_index, max_z_index + 1):
-            for zz in range(im_seg.dim[2]):
-                if zz >= min_z_index and zz <= max_z_index:
-                    # in the case of problematic segmentation (e.g., non continuous segmentation often at the extremities), display a warning but do not crash
-                    try:  # normalize the tangent vector to the centerline (i.e. its derivative)
-                        tangent_vect = self._normalize(np.array([x_centerline_deriv_rescorr[zz], y_centerline_deriv_rescorr[zz], z_centerline_deriv_rescorr[zz]]))
-                        # compute the angle between the normal vector of the plane and the vector z
-                        self.angles[zz] = np.arccos(np.vdot(tangent_vect, axis_Z))
-                    except IndexError:
-                        printv('WARNING: Your segmentation does not seem continuous, which could cause wrong estimations at the problematic slices. Please check it, especially at the extremities.', type='warning')
+        im_seg = Image(self.fname_sc)
+        nx, ny, nz, nt, px, py, pz, pt = im_seg.dim
+        data_seg = im_seg.data
+        X, Y, Z = (data_seg > 0).nonzero()
+        min_z_index, max_z_index = min(Z), max(Z)
+
+        # fit centerline, smooth it and return the first derivative (in physical space)
+        _, arr_ctl, arr_ctl_der = get_centerline(im_seg, algo_fitting='bspline', verbose=1)
+        x_centerline_fit, y_centerline_fit, z_centerline = arr_ctl
+        x_centerline_deriv, y_centerline_deriv = arr_ctl_der
+        centerline = Centerline(x_centerline_fit.tolist(), y_centerline_fit.tolist(), z_centerline.tolist(),
+                                x_centerline_deriv.tolist(), y_centerline_deriv.tolist(),
+                                np.ones_like(x_centerline_deriv).tolist())
+
+        self.angles = np.full_like(np.empty(nz), np.nan, dtype=np.double)
+
+        for iz in range(min_z_index, max_z_index + 1):
+            # in the case of problematic segmentation (e.g., non continuous segmentation often at the extremities),
+            # display a warning but do not crash
+            try:
+                # normalize the tangent vector to the centerline (i.e. its derivative)
+                tangent_vect = self._normalize(np.array(
+                    [x_centerline_deriv[iz - min_z_index] * px, y_centerline_deriv[iz - min_z_index] * py, pz]))
+
+            except IndexError:
```

Comment:
along with my previous comment, no need for try/except because you can be sure that get_centerline outputs continuous values within the vector (i.e. no "hole")-- makes the code cleaner

Reply:
Agree! Thank you